### DISCUSSION TITLE
Como executar um projeto do GitHub?

### DISCUSSION THREAD

**Discussion by Wevertonct** [frontendbr/forum#2333] [Perguntas e Respostas]
Olá pessoal! Tenho um delivery e estou a procura de um projeto que possa me ajudar com as entregas, areas de entrega, estoque, pedidos e etc... Achei esse aqui https://github.com/Mattersley/DobotAssistant bem interessante e gostaria de executá-lo,  
mas não entendo nada de programação se alguém puder me ajudar eu ficaria muito grato. Desde já agradeço a todos.

**Comment by felquis** [comment]
Eai você não manja nada?

Primeiramente, o repositório é MIT o que quer dizer que você pode usar, altera-lo a vontade. O que é impressionante.

O projeto usa nodejs. então eu faria isso:

1) bom, baixar o repositório com git clone, ai vc pode encontrar facilmente na internet algum tutorial
2) Instalaria o [nvm](https://github.com/nvm-sh/nvm#installing-and-updating) para usar uma versão do nodejs
3) O repositório tem um yarn.lock o que quer dizer que você precisa usar o [yarn](https://yarnpkg.com/getting-started/install) para instalar as dependências ele vem por padrão com o nodejs >16 basta habilita-lo.
3.a) instalar as dependências com o comando `yarn install`
3.b) agora é o famoso rodar localmente, para isso vc vai usar o comando `yarn dev` conforme definito nos [scripts do package.json](https://github.com/Mattersley/DobotAssistant/blob/0c118bb203ab0e12cd8463c55b9d42de6217f165/package.json#L47) do projeto
4) Tem algumas variáveis ambiente definidas no projeto, mas não tem nenhum .env.sample com todas elas, mas você pode ter uma noção de onde elas estão procurando por [`process.env`](https://github.com/Mattersley/DobotAssistant/search?q=process.env) em todos os arquivos.
5) Nessas variáveis de ambiente da para ter uma noção de dependências externas, pois ele usa alguns SaaS, como:
    - [Geocoding API](https://developers.google.com/maps/documentation/geocoding/get-api-key)
    - [MapBox](https://docs.mapbox.com/help/glossary/access-token/)
    - entre outros que não consigo decifrar agora rapidamente, vou deixar para vc.
6) a partir daqui fica difícil saber o que pode acontecer, então você terá que tentar rodar, ver o erro e ir decifrando o que pode ser, e como pode ser resolvido.

Isso é o melhor que posso te oferecer neste momento, espero que ajude e te desejo boa sorte!

**Reply by Wevertonct** [reply]
Puxa meu amigo, não manjo nada de programação ainda! Ajudou muito, agradeço de coração!

**Reply by Wevertonct** [reply]
@felquis e aí tudo bem!? Ontem vc me deu umas dicas sobre como executar um projeto aqui do github. Queria saber quanto vc me cobra para executar esse serviço para mim? Já te agradeço.
